### PR TITLE
Cleanup README.md

### LINKED ISSUE #1319: Review and update Mixer repo readme

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR removes the non-functional Build Status badge and alters the text slightly.

Fixes #1319

**Release note**:

```release-note-none
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1353)

### CODE REVIEW

**Comment by kyessenov** [comment]
You can point to postsubmit test jobs, like in pilot, for the badge: https://prow.istio.io/?job=pilot-postsubmit

**Comment by douglas-reid** [comment]
Ping. I'd like to close the related bug.

File: README.md [hunk 1]
```diff
@@ -2,33 +2,38 @@
...
+Card](https://goreportcard.com/badge/github.com/istio/mixer)](https://goreportcard.com/report/github.com/istio/mixer)
 [![codecov.io](https://codecov.io/github/istio/mixer/coverage.svg?branch=master)](https://codecov.io/github/istio/mixer?branch=master)
 
-Mixer provides the foundation of the Istio service mesh
-design. It is responsible for insulating Envoy and
-Istio-based services from details of the current execution
-environment, as well as to implement the control policies
-that Istio supports.
+Mixer enables extensible policy enforcement and control within the Istio service
+mesh. It is responsible for insulating the sidecar proxy (Envoy) from details of
```
Comment (geeknoid) [inline README.md]: More than just the sidecar, right? It's also for ingress.

File: README.md [hunk 2]
```diff
@@ -2,33 +2,38 @@
...
-see the Istio [contribution guidelines](https://github.com/istio/istio/blob/master/CONTRIBUTING.md).
+Please see [istio.io](https://istio.io) to learn about the overall Istio project
+and how to get in touch with us. To learn how you can contribute to any of the
+Istio components, including Mixer, please see the Istio [contribution
+guidelines](https://github.com/istio/istio/blob/master/CONTRIBUTING.md).
 
-Mixer's [developer's guide](doc/dev/development.md) presents everything
-you need to know to create, build, and test code for Mixer.
+Mixer's [developer's guide](doc/dev/development.md) presents everything you need
+to know to create, build, and test code for Mixer.
```
Comment (geeknoid) [inline README.md]: Let's add a pointer to the adapter developer's guide here.

Comment (douglas-reid) [reply README.md]: Done.

File: README.md (hunk 1 above)
Comment (douglas-reid) [reply README.md]: removed "sidecar".

**Pull request merged**